rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -93,6 +93,12 @@ bool Player::is_storage_completely_loaded() const
 
 void Player::play(const PlayOptions & options)
 {
+  rclcpp::executors::SingleThreadedExecutor exec;
+  exec.add_node(transport_node_);
+  auto node_spin_thread = std::thread([&exec]() {
+    exec.spin();
+  });
```

Comment:
I will address this in #716 but I am not convinced that's the right approach. I feel we have to separate between the player as a standalone application in which we take care of node execution and a player as an API in which the node execution will be taken care of externally.

Reply:
Yeah - I wasn't really sure what to do here because currently there is no spin on the `transport_node_`, so my publisher timer doesn't run. If you're ok with moving forward like this, then re-doing it in #716 that would work for me.